src/listeners.js

```diff
@@ -66,23 +73,74 @@ class Listener{
     }
 
     _fireListenerLogicAndSetState(val, storeSubSection, schema, ui_schema) {
-        let output = null
-        if(val["function"] == "list_adder") {
-            output = this.listNumberAdder(storeSubSection, schema, ui_schema)
-        } else if(val["function"] == "sub_list_adder") {
-            const subListPath = val["sub_list_path"]
-            output = this.subListNumberAdder(storeSubSection, subListPath)
-        } else if(val["function"] == "eval") {
-            output = this.evalForm(storeSubSection, val["eval"])
+        let output = null;
+        if (val["function"] == "list_adder") {
+            output = this.listNumberAdder(storeSubSection, schema, ui_schema);
+        } else if (val["function"] == "sub_list_adder") {
+            var subListPath = val["sub_list_path"];
+            output = this.subListNumberAdder(storeSubSection, subListPath);
+        } else if (val["function"] == "eval") {
+            output = this.evalForm(storeSubSection, val["eval"]);
+        } else if (val["function"] == "eval_list"){
+            output = this.evalFormList(storeSubSection, val["eval"], val.output_key)
         }
-        this._updateComponentStateInsideForm(val.output_key, output)
+        this._updateComponentStateInsideForm(val.output_key, output);
+    }
+
+    _updateListComponent(component_value, output) {
+        var updatesMade = false
```

Comment:
These `var`s should all be `let` or `const`. I'd reverse the `if (component_value && output){` test to exit early to make this code a little easier to read (removes a level of indentation).

Reply:
Sounds good - I think the dist version converts them all to var - we have a bit of a mix in here, but I will update the ones I added.